cmd/awscollector/main.go

```diff
@@ -51,12 +51,6 @@ func main() {
 	}
 
 	logger.SetupErrorLogger()
-
-	factories, err := defaultcomponents.Components()
```

Comment:
I'm almost sure it is not relevant, but this code was always executed as root, while by moving it here, it might be executed as the aoc user https://github.com/aws-observability/aws-otel-collector/pull/1911/files#diff-54474a2e8bb6289b7148377c1a16d0ba92e5347637414fbf582cffdc0b0b9e88R115

Reply:
I can't think of a reason why this would be an issue. Constructing the factory map does no require elevated privileges.